events/app.py

```diff
@@ -39,28 +40,24 @@ def __getattribute__(self, _):
 
 @app.route('/v1/add', methods=['POST'])
 def add_event():
-    # TODO(cmei4444): Get event information from request data
-    info = request.form['info']
-    # TODO(cmei4444): Verify that user has event editing access
-    current_time = datetime.datetime.now()
-    info = build_event_info(info, current_time)
     try:
+        info = {
+            'name': request.form['event_name'],
+            'description': request.form['description'],
+            'author': request.form['author_id'],
+            'event_time': request.form['event_time']
+        }
+        # TODO(cmei4444): Verify that user has event editing access
+        current_time = datetime.datetime.now()
+        info = build_event_info(info, current_time)
         event = Event(**info)
-    except ValueError:      # missing or extra event attributes
-        return Response(
-            status=400,
-            response="Event info was entered incorrectly.",
-        )
+    except BadRequestKeyError:      # missing event attributes
+        return "Event info was entered incorrectly.", 400
     try:
         app.config["COLLECTION"].insert_one(event.dict)
-        return Response(
-            status=201,
-        )
+        return "Event added.", 201
```

Comment:
Why do you have this in a separate try block? It looks like if the first try block raises an exception, this try block will then execute but event could be undefined so you might get an uncaught exception. Maybe it would be better to move this into the first try block and list the excepts after.

Reply:
Done in c0a375b